panther_driver/src/driver_node.py

```diff
@@ -335,22 +345,86 @@ def driverNode():
             except:
                 rospy.logwarn("Error reading rear right controller sdo")
 
+
             # position_FL / RK.encoder_resolution - > full wheel rotations
-            wheel_FL_ang_pos = 2 * math.pi * position_FL / RK.encoder_resolution  # radians
-            wheel_FR_ang_pos = 2 * math.pi * position_FR / RK.encoder_resolution
-            wheel_RL_ang_pos = 2 * math.pi * position_RL / RK.encoder_resolution
-            wheel_RR_ang_pos = 2 * math.pi * position_RR / RK.encoder_resolution
-
-            wheel_FL_ang_vel = (wheel_FL_ang_pos -
-                                wheel_FL_ang_pos_last) * dt_  # rad/s
-            wheel_FR_ang_vel = (wheel_FR_ang_pos - wheel_FR_ang_pos_last) * dt_
-            wheel_RL_ang_vel = (wheel_RL_ang_pos - wheel_RL_ang_pos_last) * dt_
-            wheel_RR_ang_vel = (wheel_RR_ang_pos - wheel_RR_ang_pos_last) * dt_
-
-            wheel_FL_ang_pos_last = wheel_FL_ang_pos
-            wheel_FR_ang_pos_last = wheel_FR_ang_pos
-            wheel_RL_ang_pos_last = wheel_RL_ang_pos
-            wheel_RR_ang_pos_last = wheel_RR_ang_pos
+            wheel_FL_ang_pos = 2.0 * math.pi * position_FL / RK.encoder_resolution / RK.gear_ratio  # radians
+            wheel_FR_ang_pos = 2.0 * math.pi * position_FR / RK.encoder_resolution / RK.gear_ratio
+            wheel_RL_ang_pos = 2.0 * math.pi * position_RL / RK.encoder_resolution / RK.gear_ratio
+            wheel_RR_ang_pos = 2.0 * math.pi * position_RR / RK.encoder_resolution / RK.gear_ratio
```

Comment:
This should be packed in a separate method, then this comment would become unnecessary

Reply:
I don't think there is a point of doing that. The code needs refactor.

Reply:
Basically, if I will make changes and move the code to functions, the rest of the package will imminently require refactor.